src/test/resources/ssl/private.key

```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```

Comment:
I think it might be worth renaming this file something like `dummy_private.key` to make it clear that we haven't accidentally checked in some real private key.

Some security / vulnerability scanners might bork at this being in the repo as well. It's worth considering if you can create one on the fly (it doesn't need to be done in java - it could be done using `make`)